kotsadm/pkg/apiserver/server.go

```diff
@@ -116,117 +116,207 @@ func Start() {
 	sessionAuthRouter.Use(handlers.RequireValidSessionMiddleware)
 
 	// Support Bundles
-	sessionAuthRouter.Path("/api/v1/troubleshoot/supportbundle/{bundleSlug}").Methods("GET").HandlerFunc(handlers.GetSupportBundle)
-	sessionAuthRouter.Path("/api/v1/troubleshoot/app/{appSlug}/supportbundles").Methods("GET").HandlerFunc(handlers.ListSupportBundles)
-	sessionAuthRouter.Path("/api/v1/troubleshoot/app/{appSlug}/supportbundlecommand").Methods("POST").HandlerFunc(handlers.GetSupportBundleCommand)
-	sessionAuthRouter.Path("/api/v1/troubleshoot/supportbundle/{bundleId}/files").Methods("GET").HandlerFunc(handlers.GetSupportBundleFiles)
-	sessionAuthRouter.Path("/api/v1/troubleshoot/supportbundle/{bundleId}/redactions").Methods("GET").HandlerFunc(handlers.GetSupportBundleRedactions)
-	sessionAuthRouter.Path("/api/v1/troubleshoot/supportbundle/{bundleId}/download").Methods("GET").HandlerFunc(handlers.DownloadSupportBundle)
-	sessionAuthRouter.Path("/api/v1/troubleshoot/supportbundle/app/{appId}/cluster/{clusterId}/collect").Methods("POST").HandlerFunc(handlers.CollectSupportBundle)
+	sessionAuthRouter.Path("/api/v1/troubleshoot/supportbundle/{bundleSlug}").Methods("GET").
+		HandlerFunc(policy.AppSupportbundleRead.Enforce(handlers.GetSupportBundle)) // TODO: appSlug
+	sessionAuthRouter.Path("/api/v1/troubleshoot/app/{appSlug}/supportbundles").Methods("GET").
+		HandlerFunc(policy.AppSupportbundleRead.Enforce(handlers.ListSupportBundles))
+	sessionAuthRouter.Path("/api/v1/troubleshoot/app/{appSlug}/supportbundlecommand").Methods("POST").
+		HandlerFunc(policy.AppSupportbundleRead.Enforce(handlers.GetSupportBundleCommand))
+	sessionAuthRouter.Path("/api/v1/troubleshoot/supportbundle/{bundleId}/files").Methods("GET").
+		HandlerFunc(policy.AppSupportbundleRead.Enforce(handlers.GetSupportBundleFiles)) // TODO: appSlug
+	sessionAuthRouter.Path("/api/v1/troubleshoot/supportbundle/{bundleId}/redactions").Methods("GET").
+		HandlerFunc(policy.AppSupportbundleRead.Enforce(handlers.GetSupportBundleRedactions)) // TODO: appSlug
+	sessionAuthRouter.Path("/api/v1/troubleshoot/supportbundle/{bundleId}/download").Methods("GET").
+		HandlerFunc(policy.AppSupportbundleRead.Enforce(handlers.DownloadSupportBundle)) // TODO: appSlug
+	sessionAuthRouter.Path("/api/v1/troubleshoot/supportbundle/app/{appId}/cluster/{clusterId}/collect").Methods("POST").
+		HandlerFunc(policy.AppSupportbundleWrite.Enforce(handlers.CollectSupportBundle))
```

Comment:
Does "write" permission imply "read" as well?  Some comments explaining how this works would be good.  It's not very obvious between Go templates and Rego...

Reply:
hmmm, no, "write" permission does not include "read" as well

Reply:
TODO: add comments